src/bot/commands/misc/avatar.js

```diff
@@ -1,24 +1,25 @@
-// Shows the avatar of the user provided
+// Shows the server avatar of the user provided
 
 import {Command} from 'yuuko';
-import {parseUser} from '../../util/discord';
+import {parseGuildMember} from '../../util/discord';
 
 const command = new Command('avatar', async (msg, args) => {
 	const MAX_AVATAR_IMAGE_SIZE = 512;
 
 	if (args.length === 0) {
-		msg.channel.createMessage(msg.author.dynamicAvatarURL('', MAX_AVATAR_IMAGE_SIZE)).catch(() => {});
+		msg.channel.createMessage(`${msg.member.avatarURL.split('?size=')[0]}?size=${MAX_AVATAR_IMAGE_SIZE}`).catch(() => {});
 		return;
 	}
-	const [member] = await parseUser(args.join(' '), msg.channel.guild, msg.author);
+	const [member] = await parseGuildMember(args[0], msg.channel.guild, msg.member);
 	if (!member) {
 		msg.channel.createMessage('Member not found, check if what you\'re using is correct or try using an ID.').catch(() => {});
 		return;
 	}
-	msg.channel.createMessage(member.dynamicAvatarURL('', MAX_AVATAR_IMAGE_SIZE)).catch(() => {});
+
+	msg.channel.createMessage(`${member.avatarURL.split('?size=')[0]}?size=${MAX_AVATAR_IMAGE_SIZE}`).catch(() => {});
```

Comment:
This URL manipulation logic seems a little complicated to have duplicated in two places. Can we break this out into a function that takes a member object and returns the processed avatar URL?

I think I'm going to PR Eris to add `dynamicAvatarURL` to `Member` objects, because members not having that built in seems like an easily fixable oversight. But even if I do, it's not likely to land in a release any time soon, so we can do our best to keep things tidy until then.

Reply:
> This URL manipulation logic seems a little complicated to have duplicated in two places. Can we break this out into a function that takes a member object and returns the processed avatar URL?
> 
> I think I'm going to PR Eris to add `dynamicAvatarURL` to `Member` objects, because members not having that built in seems like an easily fixable oversight. But even if I do, it's not likely to land in a release any time soon, so we can do our best to keep things tidy until then.

I was debating whether I should do it or not because it's only in this file and only in two places. But I'll refactor it into a function because it'll be easier to change it if needed yeah

Reply:
(resolved in 65e7a18)